### PR TITLE
Add description of Warn type class

### PR DESCRIPTION
**Pull request by LiamGoodacre**

### CODE REVIEW

**Comment by paf31** [comment]
Thanks!

File: guides/Custom-Type-Errors.md [hunk 1]
```diff
@@ -1 +1,34 @@
 See the [24 days of PureScript 2016 post](https://github.com/paf31/24-days-of-purescript-2016/blob/master/21.markdown).
+
+## The Warn type class
+
+There is a type class in `Prim` called `Warn` - which, like `Fail` is indexed by a `Symbol`.
+When the compiler solves a `Warn` constraint it will trivially solve the instance an print out the message as a `UserDefinedWarning`.
```
Comment (hdgarrood) [inline guides/Custom-Type-Errors.md]: Typo: an -> and?

Comment (hdgarrood) [inline guides/Custom-Type-Errors.md]: Also, might it be better to use normal text to say "user defined warning" here?

Comment (LiamGoodacre) [inline guides/Custom-Type-Errors.md]: Thanks, I'll fix these in the morning :)

**Pull request merged**